src/index.tsx

```diff
@@ -131,16 +139,21 @@ class ControlledWorldMap extends React.Component<Props> {
   }
 }
 
-class UncontrolledWorldMap extends React.Component<{}> {
-  state: State = { selected: null };
+class UncontrolledWorldMap extends React.Component<UncontrolledProps> {
+  state: State = { selected: [] };
 
   onMapClick = (area) => {
     this.setState(
       function() {
-        if (this.state.selected === area) {
-          return { selected: null };
+        if (this.props.multiple) {
+          const index = this.state.selected.indexOf(area);
+          if (index === -1) {
+            return this.state.selected.push(area);
+          } else {
+            return this.state.selected.splice(index, 1);
+          }
         } else {
-          return { selected: area };
+          return this.state.selected[0] === area ? this.setState({ selected: [] }) : this.setState({ selected: [area] });
```

Comment:
Since this is inside a this.setState already(starts on line 146), I think it can just return the necessary values instead of calling this.setState again.

Reply:
Good point. Fixed.